### DISCUSSION TITLE
[Auth] Show current and previous codes instead of current and next

### DISCUSSION THREAD

**Discussion by lukasbestle** [ente-io/ente#2740] [Enhancements]
The Auth UI currently displays the current and next TOTP codes. When typing the code manually on another device, the workflow becomes:

1. Look at the colorful bar to estimate if there will be enough time left to enter the current code
2. If yes, look at the current code and start typing it in on the other device
3. Otherwise ignore the current code and already use the next code
4. If the code changes mid-entry and you started with the next code, you can now continue with the current code (that was the next code before), otherwise you have a bad day and need to start over entirely

This workflow is needlessly complex. In my experience I have never needed the next code unless the remaining time for the current code was too short. To make the workflow much simpler and more enjoyable, the UI space can be used for the previous code instead, enabling the following workflow:

1. Always start looking at and typing in the current code
2. If the code changes mid-entry, continue with the previous code

**Comment by Grldk** [comment]
I would also love to be able to see the previous code. I never use the next code, and while I know of several other TOTP apps that show the previous code, I don't know of another one that shows the next code.. (But I freely admit my sample size is quite small)
